nav2_bringup/src/loc_and_nav.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2021, Samsung Research America
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#include <memory>
+#include <vector>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_map_server/map_server.hpp"
+#include "nav2_amcl/amcl_node.hpp"
+#include "nav2_controller/nav2_controller.hpp"
+#include "nav2_planner/planner_server.hpp"
+#include "nav2_recoveries/recovery_server.hpp"
+#include "nav2_lifecycle_manager/lifecycle_manager.hpp"
+#include "nav2_bt_navigator/bt_navigator.hpp"
+#include "nav2_waypoint_follower/waypoint_follower.hpp"
+
+template<typename NodeT>
+std::shared_ptr<std::thread> create_spin_thread(NodeT & node)
+{
+  return std::make_shared<std::thread>(
+    [node]() {
+      rclcpp::executors::SingleThreadedExecutor executor;
+      executor.add_node(node->get_node_base_interface());
+      executor.spin();
+      executor.remove_node(node->get_node_base_interface());
+    });
+}
+
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+
+  // localiztion nodes
```

Comment:
A bit of a better name for this file would be nice as well `composed_bringup.cpp`